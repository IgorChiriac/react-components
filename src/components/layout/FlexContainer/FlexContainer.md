The `FlexContainer` component is meant to be used as a wrapper for grid built of `FlexRow`s and `FlexCol`s.
It sets the grid's max-width and centers it horizontally.
