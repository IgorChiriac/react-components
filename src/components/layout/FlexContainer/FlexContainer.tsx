import React from 'react';
import styled from 'styled-components';
import grid from '../../../constants/grid';

export interface IFlexContainerProps {
  gutter?: number;
}

export interface IFlexContainer extends React.HTMLAttributes<HTMLDivElement>, IFlexContainerProps {}

const defaultProps: Partial<IFlexContainer> = {
  gutter: grid.gutter,
};

const StyledFlexContainer = styled.div<IFlexContainer>`
  width: 100%;
  max-width: 100%;
  padding-right: ${props => props.gutter}px;
  padding-left: ${props => props.gutter}px;
  margin-left: auto;
  margin-right: auto;
  @media (min-width: ${grid.breakpoints.s}px) {
    max-width: ${grid.width.s}px;
  }
  @media (min-width: ${grid.breakpoints.m}px) {
    max-width: ${grid.width.m}px;
  }
  @media (min-width: ${grid.breakpoints.l}px) {
    max-width: ${grid.width.l}px;
  }
  @media (min-width: ${grid.breakpoints.xl}px) {
    max-width: ${grid.width.xl}px;
  }
`;

const FlexContainer = (props: IFlexContainer) => <StyledFlexContainer {...props} />;
FlexContainer.defaultProps = defaultProps;

export default FlexContainer;
