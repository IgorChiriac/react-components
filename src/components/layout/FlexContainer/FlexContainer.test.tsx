import React from 'react';
import { render } from 'react-testing-library';
import FlexContainer from './FlexContainer';

describe('<FlexContainer />', () => {
  it('renders FlexContainer', () => {
    const { container } = render(<FlexContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
