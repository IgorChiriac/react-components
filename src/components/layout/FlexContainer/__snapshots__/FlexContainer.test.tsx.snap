// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexContainer /> renders FlexContainer 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
  padding-right: 16px;
  padding-left: 16px;
  margin-left: auto;
  margin-right: auto;
}

@media (min-width:576px) {
  .c0 {
    max-width: 540px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c0 {
    max-width: 960px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1140px;
  }
}

<div
  class="c0"
/>
`;
